yb-voyager/cmd/exportDataStatusCommand.go

```diff
@@ -0,0 +1,123 @@
+		validateExportDirFlag()
+		err := runExportDataStatusCmd()
+		if err != nil {
+			log.Errorf("Get export data status failed: %s", err)
+			fmt.Fprintf(os.Stderr, "error: %s\n", err)
+			os.Exit(1)
+		}
+	},
+}
+
+func init() {
+	exportDataCmd.AddCommand(exportDataStatusCmd)
+}
+
+type exportTableMigStatusOutputRow struct {
+	tableName string
+	status    string
+}
+
+// Note that the `export data status` is running in a separate process. It won't have access to the in-memory state
+// held in the main `export data` process.
+func runExportDataStatusCmd() error {
+	exportSchemaDoneFlagFilePath := filepath.Join(exportDir, "metainfo/flags/exportSchemaDone")
+	_, err := os.Stat(exportSchemaDoneFlagFilePath)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("cannot run `export data status` before schema export is done")
+		}
+		return fmt.Errorf("check if schema export is done: %w", err)
+	}
+
+	tableMap := make(map[string]string)
+	dataDir := filepath.Join(exportDir, "data")
+	dbTypeFlag := filepath.Join(exportDir, "metainfo", "schema", "source-db-")
+	if utils.FileOrFolderExists(dbTypeFlag + "postgresql") {
+		tableMap = GetMappingForTableNameVsTableFileName(dataDir)
+	} else if utils.FileOrFolderExists(dbTypeFlag+"mysql") || utils.FileOrFolderExists(dbTypeFlag+"oracle") {
+		files, err := filepath.Glob(filepath.Join(dataDir, "*_data.sql"))
+		if err != nil {
+			return fmt.Errorf("error while checking data directory for export data status: %v", err)
+		}
+		var fileName string
+		for _, file := range files {
+			fileName = filepath.Base(file)
+			//Sample file name: [tmp_]YB_VOYAGER_TEST_data.sql
+			if strings.HasPrefix(fileName, "tmp_") {
+				tableMap[fileName[4:len(fileName)-9]] = fileName[:len(fileName)-9]
+			} else {
+				tableMap[fileName[:len(fileName)-9]] = "tmp_" + fileName[:len(fileName)-9]
+			}
+		}
+	} else {
+		return fmt.Errorf("unable to identify source-db-type")
+	}
+
+	if len(tableMap) > 0 {
+		fmt.Printf("%-30s %-12s\n", "TABLE", "STATUS")
+	}
+	var outputRows []*exportTableMigStatusOutputRow
+	var fullTableName string
+	for tableName := range tableMap {
+		//"_" is treated as a wildcard character in regex query for Glob
+		if tableName == "postdata" {
+			continue
+		}
+		if strings.HasPrefix(tableName, "public.") {
+			fullTableName = tableName[7:]
+		} else {
+			fullTableName = tableName
+		}
+		var status string
+		if utils.FileOrFolderExists(filepath.Join(dataDir, fullTableName) + "_data.sql") {
+			status = "DONE"
+		} else if utils.FileOrFolderExists(filepath.Join(dataDir, tableMap[tableName])+"_data.sql") || utils.FileOrFolderExists(filepath.Join(dataDir, tableMap[tableName])) {
+			status = "MIGRATING"
+		} else {
+			status = "NOT_STARTED"
+		}
+
+		row := &exportTableMigStatusOutputRow{
+			tableName: tableName,
+			status:    status,
+		}
+		outputRows = append(outputRows, row)
+	}
+
+	// First sort by status and then by table-name.
+	sort.Slice(outputRows, func(i, j int) bool {
```

Comment:
I think the same comparator function is used in `importDataStatusCommand`
Can we define this comparator function separately and re-use that at both places.